### DISCUSSION TITLE
Cached query results returned

### DISCUSSION THREAD

**Discussion by rdietrick** [jsforce/jsforce#1432] [Ideas]
(So far as I can tell), there's no way to prevent getting cached query results from SFDC using conn.query().  I'm initializing some UI components with the results of a query.  If I navigate away from that page, then return to the page using the "back" button, the browser uses the cached query results on page load rather than querying SFDC a second time.

**Comment by stomita** [comment]
It seems out of JSforce's scope. You can cache the queried result within your app code.

**Comment by rdietrick** [comment]
I think you're misunderstanding.  I'm using JSForce from a browser, and I _don't_ want the browser to cache the query results.  Unfortunately, there's no way to prevent the browser from caching the query the second time it's issued.

An easy fix is to append a timestamp to the query string when making the query call to SFDC.  However, there isn't really in easy way to introduce that as an option into the API.

**Comment by stomita** [comment]
OK, understand - browser cache invalidation.

**Comment by nerdybeast** [comment]
I have noticed that if I use the 15 character Id I get a cached result but when I use an 18 character Id it seems to re-query the data.  Usually in the context of a visualforce page, you have access to the 15 character Id via a Url parameter. I would convert this to the 18 character version before you start running the query() method.

**Comment by rdietrick** [comment]
That doesn't really make sense to me, unless somehow the 18-char ID is different from request to request.  

Also, this wasn't just a query() issue.  This was happening in many different API calls (metadata, etc.).
